### ISSUE TITLE
Feature request: take CharSequence rather than String in get(..), post(...) etc

### ISSUE THREAD

**Issue by weaselmetal** [rest-assured/rest-assured#858]
We are testing our API that uses Katharsis framework with REST Assured.
The query parameter naming is a bit verbose, like ?filter[person][name]=tom.
So I created a little helper class, that allows me to construct a String representing the API-URL to test like this:

    String url = getApiResourceRoute()
                .withFilterParam("name", "Tom")
                .withFilterParam("age", 30)
                .toString();

producing:  `http://localhost:8080/api/persons?filter[persons][name]=Tom&filter[persons][age]=30`

The information that this request is about `persons` is stored in a field of the test class, in case you were wondering.

Since the `get(...)` methods and it's siblings only take types that are final, I'll always need to do something like `toString()` as my last method call in the chain. Now, if your `get(...)` methods would take a `CharSequence` instead of a `String`, I could also pass my fancy helper class directly, because it implements `CharSequence`. I would love to no not have the `toString()` there.

Would make my tests with Rest Assured even more beautiful to behold ;).

**Comment by johanhaleby** [comment]
Interesting idea. I don't think it will hurt backward compatibility to change everything to CharSequence?

**Comment by weaselmetal** [comment]
I guess it shouldn't. A `String` is a `CharSequence`. Most of the Apache `StringUtils` methods take a `CharSequence`.

**Comment by johanhaleby** [comment]
Ok I'll give it a try

**Comment by johanhaleby** [comment]
Ok that was quick try :) There are many places in Apache HTTP Client that only accept strings and not CharSequence. I think this will be too much work for what it's worth and it'll lead to casting in a lot of places. I'm sorry to say but think it's better for you to stick with `toString`.

**Comment by weaselmetal** [comment]
Thanks for trying, though I am of course very very sad now.

For what it's worth, I hoped it could be as easy as (speaking Java)

before:

    get(String path)
    {
        doSomethingWithPath(path);
    }

after:

    get(CharSequence path)
    {
        doSomethingWithPath(path.toString());
    }

well, sort-of Java, minus bloat.

**Comment by johanhaleby** [comment]
yeah, it's just that this would require a lot of changes. But if you provide a pull request that works I would probably reconsider ;)

**Comment by weaselmetal** [comment]
Gonna check. Never put my hands on groovy before :)

**Comment by weaselmetal** [comment]
Hey, one question. I guess `mvn clean test` in the project root should compile (of course) and test without errors? I'm getting errors due to missing static import of a few functions.

**Comment by johanhaleby** [comment]
Not sure about `mvn clean test` but `mvn clean install` should work. There's no contribution guide, but maybe we should write some things up? What do you suggest should be a part of this guide?

**Comment by weaselmetal** [comment]
Well, I'd say:

- Prerequisites that you'll need to get and compile the code (git, maven, JDK). A few instructions to get you set up.
- General information about the project (Languages: Java 7(?), Groovy, ...), project structure and components
- Coding conventions if you want to have some conventions, maybe an editorconfig.
- Git workflow (forking the project into your github account, providing a fix, creating a pull request, I just imagined you'd want it like that).

Basically anything that helps people getting started. Eclipse wasn't really fond of opening the project (missing gmaven plugin - dunno), so I used IntelliJ (Community Edition) to open it. I'd guess you use IntelliJ too.
